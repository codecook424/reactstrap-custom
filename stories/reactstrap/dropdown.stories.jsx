import React from "react";
import {
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
} from "reactstrap";
import { within, userEvent } from "@storybook/testing-library";

function Template(args) {
  return (
    <UncontrolledDropdown {...args}>
      <DropdownToggle caret>Dropdown</DropdownToggle>
      <DropdownMenu>
        <DropdownItem header>Header</DropdownItem>
        <DropdownItem>Some Action</DropdownItem>
        <DropdownItem text>Dropdown Item Text</DropdownItem>
        <DropdownItem disabled>Action (disabled)</DropdownItem>
        <DropdownItem divider />
        <DropdownItem>Foo Action</DropdownItem>
        <DropdownItem>Bar Action</DropdownItem>
      </DropdownMenu>
    </UncontrolledDropdown>
  );
}

export default {
  title: "Components/Dropdown",
  component: UncontrolledDropdown,
  render: Template,
  argTypes: {
    direction: {
      options: ["up", "down", "left", "right"],
      control: { type: "select" },
    },
  },
};

export const Default = {};

export const Open = {
  ...Default,
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const loginButton = await canvas.getByRole("button", {
      name: /Dropdown/i,
    });
    await userEvent.click(loginButton);
  },
};
