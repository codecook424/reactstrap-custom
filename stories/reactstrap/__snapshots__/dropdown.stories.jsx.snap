// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Dropdown Default smoke-test 1`] = `
<div class="dropdown">
  <button type="button"
          aria-haspopup="true"
          aria-expanded="false"
          class="dropdown-toggle btn btn-secondary"
  >
    Dropdown
  </button>
  <div tabindex="-1"
       role="menu"
       aria-hidden="true"
       class="dropdown-menu"
  >
    <h6 tabindex="-1"
        class="dropdown-header"
    >
      Header
    </h6>
    <button type="button"
            tabindex="0"
            role="menuitem"
            class="dropdown-item"
    >
      Some Action
    </button>
    <span tabindex="-1"
          class="dropdown-item-text"
    >
      Dropdown Item Text
    </span>
    <button type="button"
            disabled
            tabindex="-1"
            class="disabled dropdown-item"
    >
      Action (disabled)
    </button>
    <div tabindex="-1"
         class="dropdown-divider"
    >
    </div>
    <button type="button"
            tabindex="0"
            role="menuitem"
            class="dropdown-item"
    >
      Foo Action
    </button>
    <button type="button"
            tabindex="0"
            role="menuitem"
            class="dropdown-item"
    >
      Bar Action
    </button>
  </div>
</div>
`;

exports[`Components/Dropdown Open play-test 1`] = `
<div class="dropdown show">
  <button type="button"
          aria-haspopup="true"
          aria-expanded="true"
          class="dropdown-toggle btn btn-secondary"
  >
    Dropdown
  </button>
  <div tabindex="-1"
       role="menu"
       aria-hidden="false"
       class="dropdown-menu show"
       style="position: absolute; will-change: transform; top: 0px; left: 0px; transform: translate3d(0px, 38px, 0px);"
       x-placement="bottom-start"
  >
    <h6 tabindex="-1"
        class="dropdown-header"
    >
      Header
    </h6>
    <button type="button"
            tabindex="0"
            role="menuitem"
            class="dropdown-item"
    >
      Some Action
    </button>
    <span tabindex="-1"
          class="dropdown-item-text"
    >
      Dropdown Item Text
    </span>
    <button type="button"
            disabled
            tabindex="-1"
            class="disabled dropdown-item"
    >
      Action (disabled)
    </button>
    <div tabindex="-1"
         class="dropdown-divider"
    >
    </div>
    <button type="button"
            tabindex="0"
            role="menuitem"
            class="dropdown-item"
    >
      Foo Action
    </button>
    <button type="button"
            tabindex="0"
            role="menuitem"
            class="dropdown-item"
    >
      Bar Action
    </button>
  </div>
</div>
`;
